src/nvim/os/fs.c

```diff
@@ -279,13 +279,24 @@ bool os_file_is_readonly(const char *name)
   return access(name, W_OK) != 0;
 }
 
+/// Check if a file is readable.
+///
+/// @return 0 if `name` is not writable,
+/// @return 1 if `name` is writable,
+/// @return 2 for a directory which we have rights to write into.
+bool os_file_is_readable(const char *name)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_WARN_UNUSED_RESULT
+{
+  return (access(name, R_OK) == 0);
```

Comment:
@equalsraf I used `access()` because we're already using that in `os_file_is_writable()` and I didn't see any workarounds for it in #810. Ok?